Unit tests broken on x86_64 gcc 9.3

I wasn't sure what the right approach here is. Capturing the locals creates other problems. Possibly mimic how it's done in the WithoutOverwrite unit test.


[bluecheese]: /bin/x86_64-linux-gnu-g++-9 --version
x86_64-linux-gnu-g++-9 (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[bluecheese]: make
[ 23%] Built target WPEFrameworkCore
[ 28%] Built target WPEFrameworkTracing
[ 33%] Built target WPEFrameworkCryptalgo
[ 43%] Built target WPEFrameworkProtocols
[ 44%] Building CXX object Tests/unit/core/CMakeFiles/WPEFramework_test_core.dir/test_cyclicbuffer.cpp.o
/home/gladish/work/thunder/Thunder/Tests/unit/core/test_cyclicbuffer.cpp: In lambda function:
/home/gladish/work/thunder/Thunder/Tests/unit/core/test_cyclicbuffer.cpp:393:29: error: ‘bufferName’ is not captured
  393 |         CyclicBuffer buffer(bufferName,
      |                             ^~~~~~~~~~
/home/gladish/work/thunder/Thunder/Tests/unit/core/test_cyclicbuffer.cpp:385:54: note: the lambda has no capture-default
  385 |     IPTestAdministrator::OtherSideMain lambdaFunc = [](IPTestAdministrator & testAdmin) {
      |                                                      ^
/home/gladish/work/thunder/Thunder/Tests/unit/core/test_cyclicbuffer.cpp:382:20: note: ‘constexpr const char bufferName [15]’ declared here
  382 |     constexpr char bufferName[] = "cyclicbuffer04";
      |                    ^~~~~~~~~~
make[2]: *** [Tests/unit/core/CMakeFiles/WPEFramework_test_core.dir/build.make:76: Tests/unit/core/CMakeFiles/WPEFramework_test_core.dir/test_cyclicbuffer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:322: Tests/unit/core/CMakeFiles/WPEFramework_test_core.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

--- a/Source/core/CyclicBuffer.cpp
+++ b/Source/core/CyclicBuffer.cpp
@@ -195,7 +195,7 @@ namespace Core {
         }
     }
 
-    inline void CyclicBuffer::Alert()
+    void CyclicBuffer::Alert()
     {
 
         // Lock the administrator..

--- a/Tests/unit/core/test_cyclicbuffer.cpp
+++ b/Tests/unit/core/test_cyclicbuffer.cpp
@@ -382,7 +382,7 @@ TEST(Core_CyclicBuffer, WithOverwriteReversed)
     constexpr char bufferName[] = "cyclicbuffer04";
     constexpr uint32_t cyclicBufferSize = 20;
 
-    IPTestAdministrator::OtherSideMain lambdaFunc = [](IPTestAdministrator & testAdmin) {
+    auto lambdaFunc = [bufferName, cyclicBufferSize](IPTestAdministrator & testAdmin) {
         testAdmin.Sync("setup server");
 
         uint32_t result;
@@ -426,8 +426,10 @@ TEST(Core_CyclicBuffer, WithOverwriteReversed)
 
     static std::function<void (IPTestAdministrator&)> lambdaVar = lambdaFunc;
 
+    IPTestAdministrator::OtherSideMain otherSide = [](IPTestAdministrator& testAdmin ) { lambdaVar(testAdmin); };
+
     // This side (tested) acts as server
-    IPTestAdministrator testAdmin(lambdaFunc);
+    IPTestAdministrator testAdmin(otherSide);
 
     uint32_t result;
     string data;

--- a/Tests/unit/core/test_filesystem.cpp
+++ b/Tests/unit/core/test_filesystem.cpp
@@ -28,7 +28,7 @@ using namespace WPEFramework::Core;
 TEST (DISABLED_test_file, file)
 {
     File file;
-    File fileObj1("Sample.txt", true);
+    File fileObj1("Sample.txt");
     fileObj1.Create(true);
     File fileObj2(fileObj1);
     fileObj2.SetSize(150);
@@ -60,7 +60,7 @@ TEST (DISABLED_test_file, file)
 
 TEST (test_file, file_functions)
 {
-    File fileObj1("Sample2.txt", true);
+    File fileObj1("Sample2.txt");
     fileObj1.Create(true);
     char buffer[] = "Sample2.txt is moved to newFile.txt";
     fileObj1.Write(reinterpret_cast<uint8_t*>(buffer), sizeof(buffer));

--- a/Tests/unit/core/test_portability.cpp
+++ b/Tests/unit/core/test_portability.cpp
@@ -84,7 +84,7 @@ TEST(test_portability, simple_generic)
 {
     SleepS(1);
     SleepMs(1);
-    EXPECT_EQ(htonll(12345),ntohll(12345));
+    EXPECT_EQ(htonl(12345),ntohl(12345));
 
     std::thread::id parentId;
     ThreadClass object(parentId);

--- a/Tests/unit/core/test_tracing.cpp
+++ b/Tests/unit/core/test_tracing.cpp
@@ -239,8 +239,9 @@ TEST(Core_tracing, simpleTracing)
 
         while (index.Next() == true)
             if ((*index)->Enabled() == true) {
-                if ((strcmp((*index)->Module(), "Tracing")) == 0 )
+                if ((strcmp((*index)->Module(), "Tracing")) == 0 ) {
                     EXPECT_STREQ((*index)->Category(), "Information");
+                }
             }
 
         bool enabled = false;
@@ -272,8 +273,9 @@ TEST(Core_tracing, simpleTracingReversed)
 
         while (index.Next() == true)
             if ((*index)->Enabled() == true) {
-                if ((strcmp((*index)->Module(), "Tracing")) == 0 )
+                if ((strcmp((*index)->Module(), "Tracing")) == 0 ) {
                     EXPECT_STREQ((*index)->Category(), "Information");
+                }
             }
 
         bool enabled = false;

--- a/Tests/unit/core/test_workerpool.cpp
+++ b/Tests/unit/core/test_workerpool.cpp
@@ -33,7 +33,7 @@ public:
     WorkerPoolImplementation& operator=(const WorkerPoolImplementation&) = delete;
 
     WorkerPoolImplementation(const uint8_t threads, const uint32_t stackSize, const uint32_t queueSize)
-        : WorkerPool(threads, stackSize, queueSize)
+        : WorkerPool(threads, stackSize, queueSize, &_dispatcher)
     {
     }
 
@@ -54,6 +54,23 @@ public:
     {
         Core::WorkerPool::Stop();
     }
+private:
+    class Dispatcher : public Core::ThreadPool::IDispatcher {
+    public:
+      Dispatcher(const Dispatcher&) = delete;
+      Dispatcher& operator=(const Dispatcher&) = delete;
+
+      Dispatcher() = default;
+      ~Dispatcher() override = default;
+
+    private:
+      void Initialize() override { }
+      void Deinitialize() override { }
+      void Dispatch(Core::IDispatch* job) override
+        { job->Dispatch(); }
+    };
+
+    Dispatcher _dispatcher;
 };
 
 Core::ProxyType<WorkerPoolImplementation> workerpool = Core::ProxyType<WorkerPoolImplementation>::Create(2, Core::Thread::DefaultStackSize(), 8);

